src/os_maild/os_maild_client.c

```diff
@@ -762,29 +755,26 @@ void PrintTable(cJSON *item, char *printed, size_t body_size, char *tab, int cou
 
         if(body_size > log_size){
             item->string[0] = toupper(item->string[0]);
-            strncat(printed, tab, strlen(tab));
-            strncat(printed, item->string, strlen(item->string));
-            strncat(printed, delimitator, strlen(delimitator));
-            body_size -= log_size;
+            snprintf(printed + strlen(printed), body_size, "%s%s%s", tab, item->string, delimitator);
+            body_size -= strlen(tab) + strlen(item->string) + strlen(delimitator);
         }
 
         while(json_array = cJSON_GetArrayItem(item, i), json_array){
             key = cJSON_PrintUnformatted(json_array);
             log_size = strlen(key) + strlen(space);
 
             if(body_size > log_size){
-                strncat(printed, json_array->valuestring, body_size);
-                strncat(printed, space, strlen(space));
-                body_size -= log_size;
+                snprintf(printed + strlen(printed), body_size, "%s%s", json_array->valuestring, space);
```

Comment:
mmm... this is like a preexistent bug
```suggestion
                snprintf(printed + strlen(printed), body_size, "%s%s", key, space);
```